deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1StatefulSetHandler.java

```diff
@@ -312,4 +318,58 @@ public ResourceStatus status(V1StatefulSet statefulSet) {
     public int getWeight() {
         return ResourceKind.STATEFUL_SET.getWeight();
     }
+    
+    public List<String> getServiceNames(ApiClient apiClient, String selector, boolean label, String namespace)
+            throws HyscaleException {
+        return getServiceNames(getBySelector(apiClient, selector, label, namespace));
+    }
+
+    /**
+     * 
+     * @param statefulSetList
+     * @return list of service names from label of statefulset
+     */
+    public List<String> getServiceNames(List<V1StatefulSet> statefulSetList) {
+        if (statefulSetList == null) {
+            return null;
+        }
+        return statefulSetList.stream().filter(each -> {
+            return each != null && each.getMetadata() != null;
+        }).map(each -> ResourceLabelUtil.getServiceName(each.getMetadata().getLabels())).collect(Collectors.toList());
+    }
+    
+    @Override
+    public List<DeploymentStatus> getStatus(ApiClient apiClient, String selector, boolean label,
+            String namespace) {
+        try {
+            return buildStatus(getBySelector(apiClient, selector, label, namespace));
+        } catch (HyscaleException e) {
+            logger.error("Error while fetching StatefulSet with selector {} in namespace {}, error {}", selector,
+                    namespace, e.getMessage());
+        }
+        return null;
+    }
+    
+    @Override
+    public DeploymentStatus buildStatus(V1StatefulSet statefulSet) {
+        if (statefulSet == null) {
+            return null;
+        }
+        return buildStatusFromMetadata(statefulSet.getMetadata(), DeploymentStatus.Status.NOT_RUNNING);
+    }
+
+    @Override
+    public List<DeploymentStatus> buildStatus(List<V1StatefulSet> statefulSetList) {
+        if (statefulSetList == null) {
+            return null;
+        }
+        List<DeploymentStatus> statuses = new ArrayList<DeploymentStatus>();
+        statefulSetList.stream().forEach(each -> {
+            DeploymentStatus deployStatus = buildStatus(each);
+            if (deployStatus != null) {
+                statuses.add(deployStatus);
+            }
+        });
```

Comment:
Statefulset handler and deployment handler buildStatus is same right ?
give the handoff to the child and the rest can be at the parent. 
Reverse of what you're doing currently